pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/NullableSingleInputAggregationFunction.java

```diff
@@ -0,0 +1,285 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.function;
+
+import it.unimi.dsi.fastutil.floats.FloatConsumer;
+import java.util.Iterator;
+import java.util.function.Consumer;
+import java.util.function.DoubleConsumer;
+import java.util.function.Function;
+import java.util.function.LongConsumer;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.core.common.BlockValSet;
+import org.roaringbitmap.IntConsumer;
+import org.roaringbitmap.RoaringBitmap;
+
+
+public abstract class NullableSingleInputAggregationFunction<I, F extends Comparable>
```

Comment:
iiuc this class would be extensible from all null-handling aggregate features? or do we plan to only use this for mode and not the other ones?

Reply:
TL;DR: That class should be used by all aggregations once we migrate them.

I started adapting most (if not all) aggregation functions, but the PR was huge and I didn't write the test. So I decided to freeze that code. I pushed my progress to a personal branch in case I need to consult it (you can read it in https://github.com/gortiz/pinot/tree/null-aggr-bak) and I'm bringing changes from that branch into function specific branches.

Reply:
I've moved this class to https://github.com/apache/pinot/pull/12215, which is going to be the base PR of this family of PRs

Reply:
Finally I decided to move this code to https://github.com/apache/pinot/pull/12236, given #12215 is focused on the fluent framework itself. Anyway, both #12215 and #12236 should be merged (or discarded) soon.